thread/src/main/kotlin/io/github/paulgriffith/kindling/thread/MultiThreadView.kt

```diff
@@ -362,27 +349,26 @@ class MultiThreadView(
             }
         }
     }
+
     companion object {
         private val BACKGROUND = CoroutineScope(Dispatchers.Default)
 
         private fun List<ThreadDump>.toLifespanList(): List<ThreadLifespan> {
-            val map: MutableMap<Int, MutableList<Thread?>> = mutableMapOf()
-            forEachIndexed { index, threadDump ->
-                threadDump.threads.forEach {
-                    if (!map.contains(it.id)) {
-                        map[it.id] = MutableList(this.size) { null }
-                    }
-                    map[it.id]!![index] = it
-                    val condition = map[it.id]!!.distinctBy { thread -> thread?.name }.filterNotNull().size == 1
-                    require(condition) {
+            val idsToLifespans = mutableMapOf<Int, Array<Thread?>>()
+            forEachIndexed { i, threadDump ->
+                for (thread in threadDump.threads) {
+                    val array = idsToLifespans.getOrPut(thread.id) { arrayOfNulls(size) }
```

Comment:
There's no strong reason to use an array over a list here; I just personally find "array" maps cleaner to "fixed size sequence" in my head, so I understand the purpose here faster